cmd/config/apply.go

```diff
@@ -0,0 +1,154 @@
+	"github.com/armory/armory-cli/pkg/model"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+const (
+	configApplyShort   = "Sync configuration file with Armory CD-as-a-Service"
+	configApplyLong    = "Sync configuration file with Armory CD-as-a-Service"
+	configApplyExample = "armory config apply [options]"
+)
+
+type configApplyOptions struct {
+	configFile string
+}
+
+func NewConfigApplyCmd(configuration *config.Configuration) *cobra.Command {
+	options := &configApplyOptions{}
+	cmd := &cobra.Command{
+		Use:     "apply --file [<path to file>]",
+		Aliases: []string{"apply"},
+		Short:   configApplyShort,
+		Long:    configApplyLong,
+		Example: configApplyExample,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			cmdUtils.ExecuteParentHooks(cmd, args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return apply(cmd, options, configuration)
+		},
+	}
+	cmd.Flags().StringVarP(&options.configFile, "file", "f", "", "path to the configuration file")
+	err := cmd.MarkFlagRequired("file")
+	if err != nil {
+		return nil
+	}
+	return cmd
+}
+
+func apply(cmd *cobra.Command, options *configApplyOptions, configuration *config.Configuration) error {
+	payload := model.ConfigurationConfig{}
+	//in case this is running on a GitHub instance
+	gitWorkspace, present := os.LookupEnv("GITHUB_WORKSPACE")
+	_, isATest := os.LookupEnv("ARMORY_CLI_TEST")
+	if present && !isATest {
+		options.configFile = gitWorkspace + options.configFile
+	}
+	// read yaml file
+	file, err := ioutil.ReadFile(options.configFile)
+	if err != nil {
+		return fmt.Errorf("error trying to read the YAML file: %s", err)
+	}
+	cmd.SilenceUsage = true
+	// unmarshall data into struct
+	err = yaml.UnmarshalStrict(file, &payload)
+	if err != nil {
+		return fmt.Errorf("error invalid configuration object: %s", err)
+	}
+	configClient := configCmd.GetConfigClient(configuration)
+	if payload.Roles != nil {
+		err := processRoles(configClient, payload.Roles, cmd, payload.AllowAutoDelete)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func findDeletedRoles(rolesInConfigFile, existingRoles []model.RoleConfig) []string {
+	ma := make(map[string]bool, len(rolesInConfigFile))
+	var deletedRoles []string
+	for _, configRole := range rolesInConfigFile {
+		ma[configRole.Name] = true
+	}
+	for _, existinRole := range existingRoles {
+		if !ma[existinRole.Name] {
+			deletedRoles = append(deletedRoles, existinRole.Name)
+		}
+	}
+	return deletedRoles
+}
+
+func processRoles(configClient *configCmd.ConfigClient, rolesFromConfig []model.RoleConfig, cmd *cobra.Command, allowAutoDelete bool) error {
+	//get existing rolls
+	ctx, cancel := context.WithTimeout(configClient.ArmoryCloudClient.Context, time.Minute)
+	defer cancel()
+	// execute request
+	existingRoles, _, err := configClient.GetRoles(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting existing roles: %s", err)
```

Comment:
https://armory.slab.com/posts/coding-standard-proposal-package-level-error-definitions-imdq2e4w

Reply:
We do errorF all over the project so I filed a ticket to implement the coding standard proposal.
https://armory.atlassian.net/browse/CDAAS-1632